### PR TITLE
refs #1148 - test undefined on step.options.data

### PR DESCRIPTION
**Pull request by kellyjosephprice**
Previously, `eachThen` falsy values would get munged.

### CODE REVIEW

**Comment by kellyjosephprice** [comment]
This should actually be referencing #1418.

File: modules/casper.js [hunk 1]
```diff
@@ -1570,7 +1570,11 @@ Casper.prototype.runStep = function runStep(step) {
     }
     this.emit('step.start', step);
     if (this.currentResponse) {
-        this.currentResponse.data = step.options && step.options.data || null;
+        if (step.options && (undefined !== step.options.data)) {
```
Comment (paazmaya) [inline modules/casper.js]: How about using `typeof` check against `undefined`?

**Comment by istr** [comment]
I am :+1: to merge this. @mickaelandrieu @BIGjuevos

**Comment by mickaelandrieu** [comment]
Yes, why not.. this remind me we had some issues with test reporter output :/

**Comment by kellyjosephprice** [comment]
Thanks all!

**Pull request merged**